backend/shork/src/main/kotlin/internal/instruction/Sne.kt

```diff
@@ -12,10 +12,45 @@ internal class Sne(
     modifier: Modifier,
 ) : AbstractInstruction(aField, bField, addressModeA, addressModeB, modifier) {
     override fun execute(process: AbstractProcess) {
-        TODO("Not yet implemented")
+        val core = process.program.shork.memoryCore
+        val sourceAddress = resolve(process, aField, addressModeA)
+        val destinationAddress = resolve(process, bField, addressModeB)
+        val sourceInstruction = core.loadAbsolute(sourceAddress)
+        val destinationInstruction = core.loadAbsolute(destinationAddress)
+
+        val match =
+            when (modifier) {
+                Modifier.A -> {
+                    sourceInstruction.aField != destinationInstruction.aField
+                }
+                Modifier.B -> {
+                    sourceInstruction.bField != destinationInstruction.bField
+                }
+                Modifier.AB -> {
+                    sourceInstruction.aField != destinationInstruction.bField
+                }
+                Modifier.BA -> {
+                    sourceInstruction.bField != destinationInstruction.aField
+                }
+                Modifier.F -> {
+                    sourceInstruction.aField != destinationInstruction.aField &&
+                        sourceInstruction.bField != destinationInstruction.bField
+                }
+                Modifier.X -> {
+                    sourceInstruction.aField != destinationInstruction.bField &&
+                        sourceInstruction.bField != destinationInstruction.aField
+                }
+                Modifier.I -> {
+                    sourceInstruction != destinationInstruction
+                }
```

Comment:
This should only compare the opcode, modifier and fields, but not the address modes. Besides, wouldn't comparing two classes like this test for equality of the instances, not the data contained?

Reply:
In Kotlin comparing like this uses the equals method of the objects, === and !== is used for referential equality